common/src/api/external/error.rs

```diff
@@ -523,6 +523,7 @@ impl<T: ClientError> From<progenitor::progenitor_client::Error<T>> for Error {
             // our version constraints (i.e. that the call was to a newer
             // service with an incompatible response).
             progenitor::progenitor_client::Error::InvalidResponsePayload(
+                _bytes,
                 ee,
             ) => Error::internal_error(&format!(
                 "InvalidResponsePayload: {}",
```

Comment:
I think print out the bytes

Reply:
do we know why we're not just using the `Display` impl on error? these all seem close to
https://github.com/oxidecomputer/progenitor/blob/main/progenitor-client/src/progenitor_client.rs#L307-L337

Reply:
I don't know the answer to your question, no. I mostly did it for compatibility with the other arms, though it does look exactly like the display impl.

> I think print out the bytes

With `{:?}` formatting or something? I worry about size or contents, since IIUC this can be literally anything, right?

Reply:
This error would be a pretty serious problem. It would mean that we got back data that didn't match the format we expected i.e. we expected a response body of a particular type but the deserialization failed. As the comment notes, this would be reflective of a service incompatibility that would represent a pretty serious violation of an important constraint (namely that services are able to talk to each other if they're deployed together).

Yes, the output would be a pile of bytes, but if we hit this I think we'll want to know what that pile of bytes was!

Reply:
Makes sense, I'll add that to the output as well.